### PR TITLE
Go: Model `os.Args` as a `commandargs` source

### PR DESCRIPTION
**Pull request by egregius313**
`os.Args` cannot be modeled in Models as Data, so it needs a custom QL model.

#### All query authors

- [x] A change note is added if necessary. See [the documentation](https://github.com/github/codeql/blob/main/docs/change-notes.md) in this repository.
- [x] QL tests are added if necessary. See [Testing custom queries](https://docs.github.com/en/code-security/codeql-cli/using-the-advanced-functionality-of-the-codeql-cli/testing-custom-queries) in the GitHub documentation.

### CODE REVIEW

**Comment by owen-mc** [comment]
Btw I merged [this PR](https://github.com/github/codeql/pull/18034) which makes MaD work for package-level variables. Is that what was standing in the way of modeling this using MaD?

**Comment by owen-mc** [comment]
I'm guessing the test failure is because you haven't rebased since [this PR](https://github.com/github/codeql/pull/18192) was merged.

File: go/ql/test/library-tests/semmle/go/dataflow/flowsources/local/commandargs/test.expected [hunk 1]
```diff
@@ -0,0 +1,3 @@
+testFailures
+invalidModelRow
+failures
```
Comment (egregius313) [inline go/ql/test/library-tests/semmle/go/dataflow/flowsources/local/commandargs/test.expected:3]:
```suggestion
```
Remove `failures` line

**Comment by egregius313** [comment]
> Btw I merged https://github.com/github/codeql/pull/18034 which makes MaD work for package-level variables. Is that what was standing in the way of modeling this using MaD?

Oh I forgot about that change. Yes, afaik that should unblock modeling this in MaD. I'll switch it to a MaD row and check that it still works.

**Review by owen-mc — APPROVED** [review]

**Comment by owen-mc** [comment]
Are you planning on modelling more `commandargs` sources? Based on [this search](https://pkg.go.dev/search?q=flag) we should model `flag` at the very least.

**Pull request merged**